File: Problems/No.45_Jump_Game_II/jump_game_II.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        # parameters
        jumpIndex, maxJump, jump = 0, 0, 0
        i = 0
        
        # start compute
        while (jumpIndex < len(nums)-1):
            maxJump = max(maxJump, i+nums[i])
            if (i == jumpIndex):
                jumpIndex = maxJump
                jump += 1
            i += 1
                
        return jump

File: Problems/No.45_Jump_Game_II/jump_game_II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        
        // parameters
        int jumpIndex = 0, maxJump = 0, jump = 0;
        
        // start compute
        for(int i=0; jumpIndex<nums.size()-1; i++) {
            
            maxJump = max(maxJump, i+nums[i]);
            
            if(i == jumpIndex){
                
                jumpIndex = maxJump;
                jump++;
                
            }    
            
        }
        
        return jump;
        
    }
};
